22752 remaining mp metrics50 no ship

Channyboy opened the pull request:
fixes #22752


PR comments from https://github.com/OpenLiberty/open-liberty/pull/22715 to address:
- Remove `io.openliberty.mpMetrics-5.0-monitor-1.0.feature` auto feature, move the respeciive bundle activation (for the renamed `io.openliberty.microprofile.metrics.5.0.monitor.internal` )to main `mpMetrics-5.0` feature. `Monitor-1.0` is always started by `mpMetrics-xx` so no need for auto-feature.
- Remove `monitor-1.0`  condition from `io.openliberty.restfulWS3.1-mpMetrics-monitor1.0.feature`. Same reason as above.
- Appropriate Copyright
- Remove unused commented out code
- Missing condition when using `Tr.debug`
- Extra  NPE checking for `SRMetricRegistryAdapter` and `SRSharedMetricRegistriesAdapter`
- Remove `io.openliberty.smallrye.metrics/resources` folder  as it is not used.
- Rename any bundles with `internal` in the name to end with `internal` as per naming guidelines.
  - `io.openliberty.microprofile.metrics.internal.5.0` -> `io.openliberty.microprofile.metrics.5.0.internal`
  - `io.openliberty.microprofile.metrics.internal.5.0.monitor` -> `io.openliberty.microprofile.metrics.5.0.monitorinternal`
  - `io.openliberty.microprofile.metrics.internal.private.5.0` -> `io.openliberty.microprofile.metrics.5.0.private.internal`
   - `io.openliberty.microprofile.metrics.internal.public.5.0` -> `io.openliberty.microprofile.metrics.5.0.public.internal`

Additional fixes/cleanup:
- Vendor metric base unit conversion
- Include appropriate packages to omit for the REST default exception mapper (i.e. CDI related packages). In `MetricsRestfulWsEMCallbackImpl.java`
~- Delivered a disabled version of `io.openliberty.microprofile.metrics.internal.5.0_fat_tck`. Need to wait for a new RC or final release. Rely on changes in MP Metrics repository that isn't in the existing RC. The change is to accommodate changes for CDI 4.0.~
  - Deliver with https://github.com/OpenLiberty/open-liberty/issues/22808 in separate PR


Related to https://github.com/OpenLiberty/open-liberty/issues/20018

Channyboy commented:
#build

Channyboy commented:
#build

Channyboy commented:
#build
#libby

Channyboy commented:
#build
#libby

fmhwong commented on dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/smallrye/metrics/adapters/SRMetricRegistryAdapter.java:
```diff
@@ -62,7 +88,8 @@ public void addNameToApplicationMap(MetricID metricID) {
         try {
             unRegisterApplicationMetricsMethod_StringParam.invoke(srMetricRegistryObject, name);
-        } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
+        } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException
```
Generally, it is recommended to do a null check rather than catching a NPE.

Channyboy replied:
See:
https://github.com/OpenLiberty/open-liberty/pull/22772/commits/e0dc9e57a8b31ad85ee222b2467335b7e23a96b2

Channyboy commented:
#build
#libby

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0_fat_tck/publish/tckRunner/pom.xml:
```diff
@@ -0,0 +1,76 @@
+
+    <groupId>io.openliberty.microprofile.metrics</groupId>
+    <artifactId>tck.runner</artifactId>
+    <version>4.0</version>
```
Should this be 5.0?

Channyboy replied:
Yes. Done in https://github.com/OpenLiberty/open-liberty/pull/22772/commits/17faae45799eb139113f8ddf74d27ee98c686ab5

fmhwong reviewed (approved):
LGTM

Channyboy commented:
#libby

Channyboy committed:
Address remaning PR comments + other changes

- remove unnecessary auto-feature
- unnecessary prereq in auto-feature
- delete comments + delete uneeded resource folder
- copyrights
- Better debug statements and exception handling
- Unit conversion for Vendor metrics
- Renamed bundles to follow naming guidlines
- beta flag for LibraryRef configuration
- Applied formatter over classes
- TODO: FAT TCK will be delivered separately.

Channyboy commented:
#build 
#libby

Channyboy commented:
Commits squashed. Also latest squashed commit includes omitting the FAT TCK.
The disabled FAT TCK seems to be causing issues with build.
Will re-introduce in separate PR, tracked with this issue: https://github.com/OpenLiberty/open-liberty/issues/22808
The FAT TCK is disabled as it relies on changes in the MP Metric TCK which have not been made into a RC release yet. The changes to make it run with JEE10.

Channyboy committed:
Specify JAR projects to be included during packaing

Channyboy commented:
#build
#libby

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/public/mpMetrics-5.0/io.openliberty.mpMetrics-5.0.feature:
```diff
@@ -16,10 +16,11 @@ Subsystem-Name: MicroProfile Metrics 5.0
   com.ibm.websphere.appserver.anno-2.0, \
   com.ibm.websphere.appserver.monitor-1.0
 -bundles=com.ibm.ws.microprofile.metrics.common, \
- io.openliberty.io.smallrye.common.jakarta, \
```
so io.openliberty.io.smallrye.common.jakarta isn't needed?

Channyboy replied:
Nope. As per slack conversation. This was added initially to follow in line with other MP features that pulled SmallRye implementations .

However, this is not needed for `mpMetrics-5.0` to run.

jhanders34 reviewed (approved):
1 question, but approving it assuming that the answer is that it is not needed.

Channyboy commented:
Errors seen in latest build are associated with an existing build break defect and the other is related to a timeout of the server.
Those failures are not affected but the changes in this PR. Merging.
